mm2src/mm2_net/src/native_http.rs

```diff
@@ -1,12 +1,212 @@
+///     use std::time::SystemTime;
+///
+///     lazy_static! {
+///         pub(super) static ref HYPER_DANGEROUS: Client<HttpsConnector<HttpConnector>> = get_hyper_client_dangerous();
+///     }
+///
+///     fn get_hyper_client_dangerous() -> Client<HttpsConnector<HttpConnector>> {
+///         let mut config = rustls::ClientConfig::builder()
+///             .with_cipher_suites(&DEFAULT_CIPHER_SUITES)
+///             .with_safe_default_kx_groups()
+///             .with_protocol_versions(&DEFAULT_VERSIONS)
+///             .expect("inconsistent cipher-suite/versions selected")
+///             .with_root_certificates(RootCertStore::empty())
+///             .with_no_client_auth();
+///
+///         config
+///             .dangerous()
+///             .set_certificate_verifier(Arc::new(NoCertificateVerification {}));
+///
+///         let https_connector = HttpsConnectorBuilder::default()
+///             .with_tls_config(config)
+///             .https_or_http()
+///             .enable_http1()
+///             .build();
+///
+///         Client::builder().build::<_, Body>(https_connector)
+///     }
+///
+///     struct NoCertificateVerification {}
+///
+///     impl ServerCertVerifier for NoCertificateVerification {
+///         fn verify_server_cert(
+///             &self,
+///             _: &rustls::Certificate,
+///             _: &[rustls::Certificate],
+///             _: &rustls::ServerName,
+///             _: &mut dyn Iterator<Item = &[u8]>,
+///             _: &[u8],
+///             _: SystemTime,
+///         ) -> Result<ServerCertVerified, rustls::Error> {
+///             Ok(ServerCertVerified::assertion())
+///         }
+///     }
+/// }
+/// ```
+#[async_trait]
+pub trait SlurpHttpClient {
+    /// Provides a [ResponseFuture] that could be spawned and processed asynchronously
+    fn request(&self, req: Request<Body>) -> ResponseFuture;
+
+    /// Executes a POST request, returning the response status, headers and body.
+    async fn slurp_post_json(&self, url: &str, body: String) -> SlurpResult {
+        let request = Request::builder()
+            .method("POST")
+            .uri(url)
+            .header(header::CONTENT_TYPE, APPLICATION_JSON)
+            .body(body.into())?;
+        self.slurp_req(request).await
+    }
+
+    /// Executes a GET request, returning the response status, headers and body.
+    async fn slurp_url(&self, url: &str) -> SlurpResult {
+        let req = Request::builder().uri(url).body(Vec::new())?;
+        self.slurp_req(req).await
+    }
+
+    /// Executes a GET request with additional headers.
+    /// Returning the response status, headers and body.
+    async fn slurp_url_with_headers(&self, url: &str, headers: Vec<(&'static str, &'static str)>) -> SlurpResult {
+        let mut req = Request::builder();
+        let h = req.headers_mut().or_mm_err(|| {
+            SlurpError::Internal("An error occured while accessing to the request headers.".to_string())
```

Comment:
`An error occurred while accessing the request headers.`

Reply:
[done](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1910/commits/6a112a5f3a3a67bd4ac35519b1e17917a0d6b45a)